feature-libs/product-configurator/rulebased/components/attribute/header/configurator-attribute-header.component.spec.ts

```diff
@@ -790,6 +871,11 @@ describe('ConfigAttributeHeaderComponent', () => {
       testScheduler.run(() => {
         component.groupType = Configurator.GroupType.CONFLICT_GROUP;
         component.attribute.groupId = ConfigurationTestData.GROUP_ID_2;
+        const value1 = createValue('1', 'val1', false);
+        const value2 = createValue('2', 'val2', false);
+        const value3 = createValue('3', 'val3', true);
+
+        component.attribute.values = [value1, value2, value3];
```

Comment:
i believe this setup can be removed, as it is not required for this test. you are calling focusValue directly, which just delegates to the focusService without touching the attribute.

Reply:
Please, see my rework

Reply:
still there is no need for values here. We just pass on the attribute, whether it has values or not. So i believe we can removed this along with the createValue method. Still you test will pass.